frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/backing-storage-step.tsx

```diff
@@ -152,10 +152,14 @@ export const BackingStorage: React.FC<BackingStorageProps> = ({
 
   const onRadioSelect = (_, event) => {
     dispatch({ type: 'backingStorage/setType', payload: event.target.value });
-    dispatch({
-      type: 'wizard/setStepIdReached',
-      payload: 1,
-    });
+    if (stepIdReached !== 1) {
+      // Reset the wizard when a new flow is selected
+      dispatch({ type: 'wizard/setInitialState' });
+      dispatch({
+        type: 'wizard/setStepIdReached',
+        payload: 1,
+      });
+    }
```

Comment:
This is an edge case but seems like no issue in adding this logic if user selected a flow and then comes back and starts with a different flow.